### PR TITLE
Support calendar versioning

### PR DESCRIPTION
**Pull request by ohadmo**

### CODE REVIEW

**Comment by ohadmo** [comment]
rerun tests

**Comment by ohadmo** [comment]
rerun tests

File: ci/release/update-version.sh [hunk 1]
```diff
@@ -19,41 +19,41 @@
...
-RELEASE_TYPE=$1
+# Overwrite argument value to always trigger a patch release
+RELEASE_TYPE="patch"
 
 # Get current version and calculate next versions
-CURRENT_TAG=`git tag | grep -xE 'v[0-9\.]+' | sort --version-sort | tail -n 1 | tr -d 'v'`
+CURRENT_TAG=`git tag | grep -xE "v$(date '+%Y\.%m')\.[0-9]+" | sort --version-sort | tail -n 1 | tr -d 'v'`
+# In case no tag exists for this month
+if [ -z "${CURRENT_TAG}" ]; then
+  CURRENT_TAG="$(date '+%Y.%m').-1"
```
Comment (tijyojwad) [inline ci/release/update-version.sh:34]: why do we use `-1` here?

File: ci/common/test-pygenomeworks.sh [hunk 2]
```diff
@@ -51,9 +51,14 @@ pip uninstall -y pygenomeworks
...
 if [ "${COMMIT_HASH}" == "master" ]; then
+  # Official Release package
   PYGW_VERSION=$(cat ../VERSION)
 else
-  PYGW_VERSION=$(cat ../VERSION | tr -d "\n")\.dev$(date +%y%m%d) # for nightly build
+  # Development package
+  cd "$WORKSPACE"
+  bash ci/release/update-version.sh #  Promote "patch" version
+  cd "$PYGENOMEWORKS_DIR"
+  PYGW_VERSION=$(cat ../VERSION | tr -d "\n")\.dev$(date +%d) # for nightly build
```
Comment (tijyojwad) [inline ci/common/test-pygenomeworks.sh:61]: what do you think would be a good `VERSION` to use for the _upcoming_ version in the dev branch, especially if we don't know the release date?

**Review by tijyojwad — CHANGES_REQUESTED** [review]

File: ci/release/update-version.sh (hunk 1 above)
Comment (ohadmo) [reply ci/release/update-version.sh:34]: If no tags are found in the same month, it means this is the first release of this month- the next version (v2021.02.0) will be set in: `NEXT_PATCH=$((CURRENT_PATCH + 1))`. I could wrap the next commands in an if block instead.

File: ci/common/test-pygenomeworks.sh (hunk 2 above)
Comment (ohadmo) [reply ci/common/test-pygenomeworks.sh:61]: Official release -> [year][month][0 or **_current promoted_** MICRO version].
nightly release ->[year][month][0 or **_next_** MICRO version].dev[day of the month]
This is the reason `update-version.sh` is called for nightly builds.
This way our nightly packages that are released after some official release in the same month are always one MICRO version ahead. 
I assume this is okay to release a development package without an official release in some months

We should follow these [guidelines](https://packaging.python.org/guides/distributing-packages-using-setuptools/#choosing-a-versioning-scheme).

Comment (tijyojwad) [reply ci/common/test-pygenomeworks.sh:61]: documenting our offline discussion - 
```
master branch -> v2021.02.0
dev branch = next expected release data -> v2021.06.0
nightly builds = v2021.06.0.dev<todays_date>
```

**Review by tijyojwad — CHANGES_REQUESTED** [review]

**Comment by tijyojwad** [comment]
Ping @ohadmo

**Review by tijyojwad — APPROVED** [review]

**Pull request merged**